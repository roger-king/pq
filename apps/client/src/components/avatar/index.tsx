import { AvatarIcon } from './avatar';

export default AvatarIcon;
