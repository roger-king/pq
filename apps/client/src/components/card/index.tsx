export * from './question';
export * from './game';
