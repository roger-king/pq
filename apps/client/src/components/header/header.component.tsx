import React from 'react';
import { Box, Button, DropButton, Form, FormField, Header as InnerHeader, Heading } from 'grommet';
import Router, { useRouter } from 'next/router';

import { Logo } from '../logo';
import { useGlobalState } from '../../hooks/useGlobalState';

interface HeaderProps {
  gridArea: string;
}

export const Header: React.FC<HeaderProps> = (props: HeaderProps): JSX.Element => {
  const router = useRouter();
  const { isPlaying, room } = useGlobalState();
  const { gridArea } = props;

  return (
    <InnerHeader gridArea={gridArea} pad={{ left: '50px', right: '50px' }}>
      <Heading level="3" onClick={() => Router.push('/')} style={{ cursor: 'pointer' }}>
        <Logo />
      </Heading>
      {/* TODO: Add room name when joined */}
      <Heading>{room}</Heading>
      {isPlaying ? (
        <Box width="50px" height="20px" />
      ) : (
        <DropButton
          label="Join Game"
          alignSelf="center"
          dropContent={
            <Box pad="large" background="brand">
              <Form
                onSubmit={(e) => {
                  const { code }: any = e.value;
                  router.push(`/pq/${code}`);
                }}
              >
                <FormField name="code" placeholder="Room Code" maxLength={12} minLength={12} required />
                <Button type="submit" label="Join" />
              </Form>
            </Box>
          }
          dropAlign={{ top: 'bottom', right: 'right' }}
        />
      )}
    </InnerHeader>
  );
};
