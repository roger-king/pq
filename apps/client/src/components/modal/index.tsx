import React, { PropsWithChildren } from 'react';
import { Box, Heading, Layer } from 'grommet';

export interface BaseModalProps {
  title: string;
  onClose(): void;
}

const Modal: React.FC<PropsWithChildren<BaseModalProps>> = (props: PropsWithChildren<BaseModalProps>) => {
  const { children, title } = props;

  return (
    <Layer modal>
      <Box fill align="center" justify="center" pad="medium" background="brand">
        <Box direction="row" align="center" justify="between" fill>
          <Heading level="3">{title}</Heading>
        </Box>
        {children}
      </Box>
    </Layer>
  );
};

export default Modal;
