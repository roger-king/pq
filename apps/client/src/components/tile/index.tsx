import { Tile } from './tile.component';

export default Tile;
