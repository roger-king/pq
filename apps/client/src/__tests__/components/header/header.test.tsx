import React from 'react';
import renderer from 'react-test-renderer';

import { Header } from '../../../components/header';

describe('<Header />', () => {
  test('should display Header', async () => {
    const tree = renderer.create(<Header gridArea="header" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
