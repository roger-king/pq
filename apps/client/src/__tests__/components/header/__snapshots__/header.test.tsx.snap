// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should display Header 1`] = `
<header
  className="StyledBox-sc-13pk1d4-0 eTdmLy"
>
  <h3
    className="StyledHeading-sc-1rdh4aw-0 XVrap"
    onClick={[Function]}
    style={
      Object {
        "cursor": "pointer",
      }
    }
  >
    <div
      className="StyledBox-sc-13pk1d4-0 cIkaNd"
    >
      <h2
        className="StyledHeading-sc-1rdh4aw-0 frzykv"
      >
        PQ
      </h2>
    </div>
  </h3>
  <div
    className="StyledBox__StyledBoxGap-sc-13pk1d4-1 ffHQai"
  />
  <h1
    className="StyledHeading-sc-1rdh4aw-0 fmEWrv"
  >
    
  </h1>
  <div
    className="StyledBox__StyledBoxGap-sc-13pk1d4-1 ffHQai"
  />
  <div
    className="StyledBox-sc-13pk1d4-0 fYCwfB"
  />
</header>
`;
